lib/internal/url.js

```diff
@@ -1308,11 +1308,17 @@ function originFor(url, base) {
 }
 
 function domainToASCII(domain) {
+  if (arguments.length < 1)
```

Comment:
Since `url.host = undefined` actually sets the domain to "undefined"...it makes sense for `domainTo*` too so `arguments.length` is needed